tools/handshake-runner/Simulation.cpp

```diff
@@ -45,127 +51,298 @@ static void fatalValueError(StringRef reason, T &value) {
   llvm::report_fatal_error(err);
 }
 
-void executeOp(mlir::ConstantIndexOp op, std::vector<Any> &in,
-               std::vector<Any> &out) {
+void debugArg(const std::string &head, mlir::Value op, const APInt &value,
+              double time) {
+  LLVM_DEBUG(dbgs() << "  " << head << ":  " << op << " = " << value
+                    << " (APInt<" << value.getBitWidth() << ">) @" << time
+                    << "\n");
+}
+
+void debugArg(const std::string &head, mlir::Value op, const APFloat &value,
+              double time) {
+  LLVM_DEBUG(dbgs() << "  " << head << ":  " << op << " = ";
+             value.print(dbgs()); dbgs() << " ("
+                                         << "float"
+                                         << ") @" << time << "\n");
+}
+
+void debugArg(const std::string &head, mlir::Value op, const Any &value,
+              double time) {
+  if (any_isa<APInt>(value)) {
+    debugArg(head, op, any_cast<APInt>(value), time);
+  } else if (any_isa<APFloat>(value)) {
+    debugArg(head, op, any_cast<APFloat>(value), time);
+  } else if (any_isa<unsigned>(value)) {
+    // Represents an allocated buffer.
+    LLVM_DEBUG(dbgs() << "  " << head << ":  " << op << " = Buffer "
+                      << any_cast<unsigned>(value) << "\n");
+  } else {
+    llvm_unreachable("unknown type");
+  }
+}
+
+Any readValueWithType(mlir::Type type, std::string in) {
+  std::stringstream arg(in);
+  if (type.isIndex()) {
+    int64_t x;
+    arg >> x;
+    int64_t width = INDEX_WIDTH;
+    APInt aparg(width, x);
+    return aparg;
+  } else if (type.isa<mlir::IntegerType>()) {
+    int64_t x;
+    arg >> x;
+    int64_t width = type.getIntOrFloatBitWidth();
+    APInt aparg(width, x);
+    return aparg;
+    // } else if (type.isF16()) {
```

Comment:
remove commented out code, or uncomment?

Reply:
Will do; there seems to be quite a bit of lingering commented out pieces of code around these parts...